src/js/modules/users/selectors.js

```diff
@@ -1,46 +1,84 @@
 import { createSelector } from "reselect";
+import appHistory from "../../tools/appHistory";
 
 export const getUsers = state => state.users.users;
 export const getRoles = state => state.users.roles;
 export const getUserRoles = state => state.users.userRoles;
 
 const getRequestedContributorSlug = (state, props) => props.match.params.contributor_slug;
+const getRequestedIllustratorSlug = (state, props) => props.match.params.illustrator_slug;
+const getRequestedPhotographerSlug = (state, props) => props.match.params.photographer_slug;
+
 const getRoleFromProps = (state, props) => props.role;
-const getUsersResponse = state => state.users.response;
+
+const CONTRIBUTOR_ROLE_ID = 0;
```

Comment:
put these in the constants file. Not selectors

Reply:
Actually....hard code them into state. Then create selectors that get the role constants, then you can just compose the selectors accordingly. If you do it intelligently you might even shorten the code by a lot